packages/opentelemetry-sdk-trace-base/src/export/BatchSpanProcessorBase.ts

```diff
@@ -181,7 +181,13 @@ export abstract class BatchSpanProcessorBase<T extends BufferConfig>
         // Reset the finished spans buffer here because the next invocations of the _flush method
         // could pass the same finished spans to the exporter if the buffer is cleared
         // outside the execution of this callback.
-        const spans = this._finishedSpans.splice(0, this._maxExportBatchSize);
+        let spans: ReadableSpan[];
+        if (this._finishedSpans.length <= this._maxExportBatchSize) {
+          spans = this._finishedSpans;
+          this._finishedSpans = [];
+        } else {
+          spans = this._finishedSpans.splice(0, this._maxExportBatchSize);
+        }
```

Comment:
The splice method will change the existing array, and it seems that it will include some resizing, which is expensive. But at very least if will return new array with deleted elements and this will for sure have to be allocated. My idea was to add fast path when you have `maxQueueSize === maxExportBatchSize` to avoid paying this price. So basically add to buffer until its full and then clear it at once without  any manipulations.

Reply:
Got it :+1: